pkg/scaler/scaling_test.go

```diff
@@ -561,17 +512,13 @@ var _ = Describe("Node scaling tests", func() {
 						Expect(found).To(BeTrue())
 						return nil
 					}).Times(2),
-				mockKubeClient.EXPECT().List(gomock.Any(), gomock.Any()).SetArg(1, *nodes),
 				mockKubeClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).SetArg(1, *upgradeMachines),
```

Comment:
This should not be removed, nor should the nodes variables declared above it be removed. I think there is a misunderstanding happening of what the `gomock.InOrder` function is performing. 

The list of calls happening in `gomock.InOrder` are mocking calls that the function would otherwise make to the kube API. When running these test cases there is no kube API for the code to call, so the tests mock it out and act like a fake client that returns responses. 

There is a big block of code in `ensureScaleDownNodes` which runs _before_ the change that this PR is making, which is involved around retrieving nodes and executing drain strategies on them. None of that code is changing, so the mock calls still need to be there. 

For instance, this line still gets called:
https://github.com/anjoshi24/managed-upgrade-operator/blob/4b27f7710a75ebee0e88b35fb39a5c1e5fb00825/pkg/scaler/machineSetScaler.go#L100
which results in a `List` of Nodes. It is still necessary to have the test set up a mock for this call.

Reply:
Added the required list call for nodes.